### PR TITLE
Implement AnnounceOnBuildInfo.OnlyToOwner

### PR DESCRIPTION
**Pull request by IceReaper**
Implement a property to make the voice only play for the particular owner instead of global.

### CODE REVIEW

**Comment by abcdefg30** [comment]
See https://github.com/OpenRA/OpenRA/pull/11093#issuecomment-208032565.

**Comment by GraionDilach** [comment]
Also see https://github.com/AttacqueSuperior/Engine/issues/68.

**Comment by IceReaper** [comment]
Updated!

File: OpenRA.Mods.Common/Traits/Sound/AnnounceOnBuild.cs [hunk 1]
```diff
@@ -19,6 +19,9 @@ public class AnnounceOnBuildInfo : ITraitInfo
 		[Desc("Voice to use when built/trained.")]
 		[VoiceReference] public readonly string Voice = "Build";
 
+		[Desc("Player stances who can hear this voice. Stance Neutral applies to the local player only.")]
```
[the remark this answers is not in the document]
Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Sound/AnnounceOnBuild.cs]: I would default it to `true`, however.

**Review by pchote — DISMISSED** [review]

**Review by abcdefg30 — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/Sound/AnnounceOnBuild.cs [hunk 2]
```diff
@@ -33,7 +39,10 @@ public AnnounceOnBuild(Actor self, AnnounceOnBuildInfo info)
 
 		void INotifyBuildComplete.BuildingComplete(Actor self)
 		{
-			self.PlayVoice(info.Voice);
+			if (info.ValidStances.HasStance(self.Owner.Stances[self.World.LocalPlayer]))
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Sound/AnnounceOnBuild.cs]: Actually, this crashes when `self.World.LocalPlayer` is `null`. If that is the case we might want to check `RenderPlayer`.

Comment (pchote) [reply OpenRA.Mods.Common/Traits/Sound/AnnounceOnBuild.cs]: will also need to handle the case where both are null.

**Comment by IceReaper** [comment]
Updated!

**Review by abcdefg30 — APPROVED** [review]
Didn't test the latest set of changes but it looks good code wise.

**Pull request merged**